adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -666,7 +714,7 @@ def _wait_for_msg(self, timeout=30):
             sz -= 0x02
         msg = self._sock.recv(sz)
         if self.on_message is not None:
-            self.on_message(self, topic, str(msg, "utf-8"))
+            self._handle_on_message(self, topic, str(msg, "utf-8"))
```

Comment:
The above check `if self.on_message is not None:` should be removed. The `_handle_on_message` method will check to see if `on_message` is set only if the topic doesn't match a handler.

Otherwise you always have to set `on_message`, even if you have added handles for. the topics

Reply:
Ok, I'm going to remove this check and test on h/w to confirm.